components/tools/OmeroPy/src/omero/gateway/__init__.py

```diff
@@ -7454,15 +7458,26 @@ def getChannels(self):
         # E.g. ConcurrencyException (no rendering engine): load channels by
         # hand, use pixels to order channels
         else:
-            pid = self.getPixelsId()
-            params = omero.sys.Parameters()
-            params.map = {"pid": rlong(pid)}
-            query = ("select p from Pixels p join fetch p.channels as c "
-                     "join fetch c.logicalChannel as lc where p.id=:pid")
-            pixels = self._conn.getQueryService().findByQuery(
-                query, params, self._conn.SERVICE_OPTS)
-            return [ChannelWrapper(self._conn, c, idx=n, re=self._re, img=self)
-                    for n, c in enumerate(pixels.iterateChannels())]
+            return self.getChannelsNoRE()
+
+    def getChannelsNoRE(self):
```

Comment:
Agreed that would be ideal.
I guess I was a little reluctant to mess with the class-based ```@assert_re``` decorator on image.getChannels() that would have to be removed and the same functionality applied within the method itself only if NoRE is True.
But I'll see if this is possible without breaking anything.